src/analysis/normed_space/star/continuous_functional_calculus.lean

```diff
@@ -0,0 +1,275 @@
+  -- goose chase when trying to apply `spectrum.gelfand_transform_eq`.
+  letI := elemental_star_algebra.complex.normed_algebra a,
+  unfreezingI { rcases subsingleton_or_nontrivial A },
+  { simp only [spectrum.of_subsingleton, set.empty_subset] },
+  { set a' : elemental_star_algebra ℂ a := ⟨a, self_mem ℂ a⟩,
+    refine (spectrum.subset_star_subalgebra (star a' * a')).trans _,
+    rw [←spectrum.gelfand_transform_eq (star a' * a'), continuous_map.spectrum_eq_range],
+    rintro - ⟨φ, rfl⟩,
+    rw [gelfand_transform_apply_apply ℂ _ (star a' * a') φ, map_mul φ, map_star φ],
+    rw [complex.eq_coe_norm_of_nonneg star_mul_self_nonneg, ←map_star, ←map_mul],
+    exact ⟨complex.zero_le_real.2 (norm_nonneg _),
+      complex.real_le_real.2 (alg_hom.norm_apply_le_self φ (star a' * a'))⟩, }
+end
+
+variables {a}
+
+/-- This is the key lemma on the way to establishing spectral permanence for C⋆-algebras, which is
+established in `star_subalgebra.spectrum_eq`. This lemma is superseded by
+`star_subalgebra.coe_is_unit`, which does not require an `is_star_normal` hypothesis and holds for
+any closed star subalgebra. -/
+lemma elemental_star_algebra.is_unit_of_is_unit_of_is_star_normal (h : is_unit a) :
+  is_unit (⟨a, self_mem ℂ a⟩ : elemental_star_algebra ℂ a) :=
+begin
+  /- Sketch of proof: Because `a` is normal, it suffices to prove that `star a * a` is invertible
+  in `elemental_star_algebra ℂ a`. For this it suffices to prove that it is sufficiently close to a
+  unit, namely `algebra_map ℂ _ ‖star a * a‖`, and in this case the required distance is
+  `‖star a * a‖`. So one must show `‖star a * a - algebra_map ℂ _ ‖star a * a‖‖ < ‖star a * a‖`.
+  Since `star a * a - algebra_map ℂ _ ‖star a * a‖` is selfadjoint, by a corollary of Gelfand's
+  formula for the spectral radius (`is_self_adjoint.spectral_radius_eq_nnnorm`) it's norm is the
+  supremum of the norms of elements in its spectrum (we may use the spectrum in `A` here because
+  the norm in `A` and the norm in the subalgebra coincide).
+
+  By `spectrum_star_mul_self_of_is_star_normal`, the spectrum (in the algebra `A`) of `star a * a`
+  is contained in the interval `[0, ‖star a * a‖]`, and since `a` (and hence `star a * a`) is
+  invertible in `A`, we may omit `0` from this interval. Therefore, by basic spectral mapping
+  properties, the spectrum (in the algebra `A`) of `star a * a - algebra_map ℂ _ ‖star a * a‖` is
+  contained in `[0, ‖star a * a‖)`. The supremum of the (norms of) elements of the spectrum must be
+  *strictly* less that `‖star a * a‖` because the spectrum is compact, which completes the proof. -/
+
+  /- We may assume `A` is nontrivial. It suffices to show that `star a * a` is invertible in
+  `elemental_star_algebra ℂ a` because `a` is normal, so `star a * a` is invertible if and only if
+  both `star a` and `a` are invertible, but `star a` is invertible if and only if `a` is. -/
```

Comment:
hmmm, I would have said that's the *nontrivial* direction (because it requires commutativity of `a` with `star a`), whereas the other direction is always true. Nevertheless, I will edit the comment to be more precise.